spec/services/service_offering/add_to_portfolio_item_spec.rb

```diff
@@ -47,6 +47,21 @@
     end
   end
 
+  context "when there is no icon" do
+    let(:topology_service_offering) do
+      offering = fully_populated_service_offering
+      offering.service_offering_icon_id = nil
+      offering
```

Comment:
Is there were I get to suggest using `tap`?  Not sure if you will like this one-liner better.
```ruby
fully_populated_service_offering.tap { |so| so.service_offering_icon_id = nil }
```

Reply:
Oh wow yeah thats the perfect usage of tap. Changed!